azure/services/aso/service.go

```diff
@@ -31,6 +32,9 @@ type Service[T deepCopier[T], S Scope] struct {
 
 	Scope S
 	Specs []azure.ASOResourceSpecGetter[T]
+	// ListFunc is used to enumerate ASO resources to determine which stale ones to delete. This behavior may
+	// be skipped for a service by leaving this field nil.
```

Comment:
Why would a service ever not want to implement this?

Reply:
Sorry, just finished an initial pass at adding my own comments and am just getting to yours. I explained that here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4506#discussion_r1465460586